homeassistant/components/subaru/config_flow.py

```diff
@@ -0,0 +1,167 @@
+"""Config flow for Subaru integration."""
+from datetime import datetime
+import logging
+
+from subarulink import (
+    Controller as SubaruAPI,
+    InvalidCredentials,
+    InvalidPIN,
+    SubaruException,
+)
+import voluptuous as vol
+
+from homeassistant import config_entries, core
+from homeassistant.const import (
+    CONF_DEVICE_ID,
+    CONF_PASSWORD,
+    CONF_PIN,
+    CONF_SCAN_INTERVAL,
+    CONF_USERNAME,
+)
+from homeassistant.core import callback
+from homeassistant.helpers import aiohttp_client, config_validation as cv
+
+from .const import (
+    CONF_HARD_POLL_INTERVAL,
+    DEFAULT_HARD_POLL_INTERVAL,
+    DEFAULT_SCAN_INTERVAL,
+    DOMAIN,
+    MIN_HARD_POLL_INTERVAL,
+    MIN_SCAN_INTERVAL,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@callback
+def configured_instances(hass):
+    """Return a set of configured Subaru instances."""
+    return {entry.title for entry in hass.config_entries.async_entries(DOMAIN)}
+
+
+class SubaruConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Subaru."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    async def async_step_user(self, user_input=None):
+        """Handle the start of the config flow."""
+        error = None
+
+        if not user_input:
+            return self.async_show_form(
+                step_id="user",
+                data_schema=vol.Schema(
+                    {
+                        vol.Required(CONF_USERNAME): str,
+                        vol.Required(CONF_PASSWORD): str,
+                        vol.Required(CONF_PIN): str,
+                    }
+                ),
+            )
```

Comment:
This isn't needed anymore - you reverse the condition, and then use a single show form (with or without errors) at the bottom

Reply:
Change incorporated